nibabel/freesurfer/io.py

```diff
@@ -388,18 +504,31 @@ def write_annot(filepath, labels, ctab, names):
         RGBA + label id colortable array.
     names : list of str
         The names of the labels. The length of the list is n_labels.
+    fill_ctab : {True, False} optional
+        If True, the annotation values for each vertex  are automatically
+        generated. In this case, the provided `ctab` may have shape
+        (n_labels, 4) or (n_labels, 5) - if the latter, the final column is
+        ignored.
     """
     with open(filepath, "wb") as fobj:
-        dt = ">i4"
+        dt = _ANNOT_DT
         vnum = len(labels)
 
         def write(num, dtype=dt):
             np.array([num]).astype(dtype).tofile(fobj)
 
         def write_string(s):
+            s = s.encode() + b'\00'
```

Comment:
Maybe

```Python
s = (s if isinstance(s, bytes) else s.encode()) + b'\x00'
```

Also, note the `x` in `'\x00'`. `b'\00' == b'\x000'`.

Reply:
Oops .. I made sure to test on py2 before pushing, but at the cost of forgetting to test on py3!